### DISCUSSION TITLE
pageList에서의 isSkipPage 필요성

### DISCUSSION THREAD

**Discussion by herrakam** [eRentronic/eRentronic-Client#81] [아이디어]
### 현 상황
- `isSkipPage:boolean`을 필수 props로 받음
-  true라면 `...` 이 추가
- false라면 모든 번호를 보여줌

### 의견 제시
- 페이지에서 스킵 기능을 활성화할지 말지를 받는것 대신 보여줘야 되는 페이지 수 기준을 props로 받는 건 어떨까?
### 개선시 로직
- 기본적으로는 모든 번호를 보여줌
- 보여줘야 되는 페이지 수가 기준(props로 받은 값)보다 커지면 `...`가 추가됨(스킵 기능 활성화)

**Comment by godhyeongman** [comment]
갠적으로는 isSkipPage를 optional하게 받아서 사용하면 좋을 것 같습니다. 이유는 저번에 말한것 처럼 페이지 리스트를 진짜 전부 보여주고 싶은 경우를 가정 했을때이기 때문입니다. 상황을 가정하면 1 부터 7까지리를 보여주고 싶은데 저절로 스킵 되는걸 조절 할 수 가 없잖아요! 기본 설정을 전달 받은 데이터를 계산하여 판단하고 스킵하고, 만약 SkipPage가 false 값이 왔을때는 전부 보여준다는 어떠실까요?

**Reply by herrakam** [reply]
그 문제는 마지막 페이지값을 이 기준 props 위치로  넘겨주면 해결 될 거 같아요
  보여줘야 될 페이지 수가 기준보다 커질 때 스킵로직이 활성화되는데, 마지막 페이지값을 넣어주면 보여줘야 될 페이지 수가 기준 이하이기 때문에 스킵로직이 비활성화되서 모든 페이지를 보여주게 됩니다! 스킵을 쓰는 경우가 아닌 경우보다 많을 것 같지만 기본값으로 마지막 페이지 값을 넘겨주게 되면 기본동작을 모든 페이지를 보여주게도 구현할 수 있겠네요

**Reply by godhyeongman** [reply]
좋네요 스토리북에 문서로 마지막 페이지시 full page라고 사용법이 명시되어 있으면 사용하기 수월할것 같아요~

**Comment by herrakam** [comment]
isSkipPage 대신 skipNumber를 받아서, 이 수보다 보여줘야 될 페이지가 많으면 스킵 로직이 실행되는 방향으로 결정
